components/public-api/gitpod/v1/workspaces.proto

```diff
@@ -0,0 +1,322 @@
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/field_mask.proto";
+import "google/rpc/status.proto";
+import "gitpod/v1/pagination.proto";
+
+service WorkspacesService {
+
+    // ListWorkspaces enumerates all workspaces belonging to the authenticated user.
+    rpc ListWorkspaces(ListWorkspacesRequest) returns (ListWorkspacesResponse) {}
+
+    // GetWorkspace returns a single workspace.
+    rpc GetWorkspace(GetWorkspaceRequest) returns (GetWorkspaceResponse) {}
+
+    // CreateAndStartWorkspace creates a new workspace and starts it.
+    rpc CreateAndStartWorkspace(CreateAndStartWorkspaceRequest) returns (CreateAndStartWorkspaceResponse) {}
+
+    // StartWorkspace starts an existing workspace.
+    rpc StartWorkspace(StartWorkspaceRequest) returns (StartWorkspaceResponse) {}
+
+    // GetRunningWorkspaceInstance returns the currently active instance of a workspace.
+    // Errors:
+    //   FAILED_PRECONDITION: if a workspace does not a currently active instance
+    //
+    rpc GetActiveWorkspaceInstance(GetActiveWorkspaceInstanceRequest) returns (GetActiveWorkspaceInstanceResponse) {}
+
+    // GetWorkspaceInstanceOwnerToken returns the owner token of a workspace instance.
+    // Note: the owner token is not part of the workspace instance status so that we can scope its access on the
+    //       API function level.
+    rpc GetWorkspaceInstanceOwnerToken(GetWorkspaceInstanceOwnerTokenRequest) returns (GetWorkspaceInstanceOwnerTokenResponse) {}
+
+    // ListenToWorkspaceInstance listens to workspace instance updates.
+    rpc ListenToWorkspaceInstance(ListenToWorkspaceInstanceRequest) returns (stream ListenToWorkspaceInstanceResponse) {}
+
+    // ListenToImageBuildLogs streams (currently or previously) running workspace image build logs
+    rpc ListenToImageBuildLogs(ListenToImageBuildLogsRequest) returns (stream ListenToImageBuildLogsResponse) {}
+
+    // StopWorkspace stops a running workspace (instance).
+    // Errors:
+    //   NOT_FOUND:           the workspace_id is unkown
+    //   FAILED_PRECONDITION: if there's no running instance
+    rpc StopWorkspace(StopWorkspaceRequest) returns (stream StopWorkspaceResponse) {}
+
+}
+
+message ListWorkspacesRequest {
+    Pagination pagination = 1;
+    google.protobuf.FieldMask field_mask = 2;
+}
+message ListWorkspacesResponse {
+    message WorkspaceAndInstance {
+        Workspace result = 1;
+        WorkspaceInstance last_active_instances = 2;
+    }
+
+    google.rpc.Status response_status = 1;
+
+    string next_page_token = 2;
+
+    repeated WorkspaceAndInstance result = 3;
+}
+
+message GetWorkspaceRequest {
+    string workspace_id = 1;
+}
+message GetWorkspaceResponse {
+    google.rpc.Status response_status = 1;
+
+    Workspace result = 2;
+}
+
+message CreateAndStartWorkspaceRequest {
+    string idempotency_token = 1;
+
+    oneof source {
+        string context_url = 2;
+        string prebuild_id = 3;
```

Comment:
Hm, this feels sligthly odd... :thinking: cannot pinpoint, though. Have to sleep over it.

Reply:
Why so? 
As we now expose the prebuild ids on the dashboard, it would be nice to actually make use of them instead of going though all of the indirections.